libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -877,36 +926,47 @@ void AP_ADSB::set_callsign(const char* str, const bool append_icao)
     if (append_icao) {
         snprintf(&out_state.cfg.callsign[4], 5, "%04X", unsigned(out_state.cfg.ICAO_id % 0x10000));
     }
+    return true;
 }
 
 
 void AP_ADSB::push_sample(const adsb_vehicle_t &vehicle)
 {
-    samples.push(vehicle);
+    _samples.push(vehicle);
 }
 
 bool AP_ADSB::next_sample(adsb_vehicle_t &vehicle)
 {
-    return samples.pop(vehicle);
+    return _samples.pop(vehicle);
 }
 
 void AP_ADSB::handle_message(const mavlink_channel_t chan, const mavlink_message_t &msg)
 {
     switch (msg.msgid) {
-    case MAVLINK_MSG_ID_ADSB_VEHICLE:
-        handle_vehicle(msg);
+    case MAVLINK_MSG_ID_ADSB_VEHICLE: {
+        adsb_vehicle_t vehicle {};
+        mavlink_msg_adsb_vehicle_decode(&msg, &vehicle.info);
+        vehicle.last_update_ms = AP_HAL::millis() - (vehicle.info.tslc * 1000);
```

Comment:
should be:
AP_HAL::millis() - uint32_t(vehicle.info.tslc * 1000U)

Reply:
fixed